apm-sniffer/apm-sdk-plugin/okhttp-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/okhttp/v3/RealCallInterceptor.java

```diff
@@ -70,10 +69,6 @@ public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allAr
         SpanLayer.asHttp(span);
 
         Field headersField = Request.class.getDeclaredField("headers");
-        Field modifiersField = Field.class.getDeclaredField("modifiers");
-        modifiersField.setAccessible(true);
-        modifiersField.setInt(headersField, headersField.getModifiers() & ~Modifier.FINAL);
-
         headersField.setAccessible(true);
         Headers.Builder headerBuilder = request.headers().newBuilder();
```

Comment:
Note, the existing code creates a new builder, but never read the existing headers, if the client has meaning headers, such as token, they would be lost, right?

Reply:
I know this is not related to this fix. You could consider to check these 3 plugins, to submit issue or another PR to fox this.

Reply:
first question : request.headers().newBuilder(); 
this code read existing headers, and next step put new values in it.

2: there just 3 block code use this way to change modifiers  in project.

Reply:
Thanks for clearance. Good to me.